snapcraft/tests/fixture_setup.py

```diff
@@ -972,3 +975,185 @@ def setUp(self):
         with open(os.path.join(self.path, 'snapcraft.yaml'),
                   'w') as snapcraft_yaml_file:
             yaml.dump(self.data, snapcraft_yaml_file)
+
+
+class ObjectSpy(fixtures.Fixture):
+
+    def __init__(self, target, cls):
+        super().__init__()
+
+        self._target = target
+        self._cls = cls
+
+    def _setUp(self):
+        patcher = mock.patch(self._target, wraps=self._cls)
+        self.spy = patcher.start()
+        self.addCleanup(patcher.stop)
+
+        for member in inspect.getmembers(self._cls,
+                                         predicate=inspect.isfunction):
+            name = member[0]
```

Comment:
can you call these `member_name` and `member_value`, some lines below I lost context